src/Commands/ClearCacheFiles.php

```diff
@@ -0,0 +1,128 @@
+            ->setName('fop:clear-cache')
+            ->setDescription('Replace the cache directory with an empty one.')
+            ->setHelp('This command allows you to quickly remove the cache files.');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        try {
+            // first : disable event dispatcher otherwise errors will happen
+            $this->getApplication()->setDispatcher(new EventDispatcher());
+            $this->processChecks();
+            $this->deleteOldCacheDirectory(); // may exist if this command failed before
+            $this->renameCurrentCacheDirectory();
+            $this->createNewCacheDirectory(); // probably not needed
+            $io->success('New Empty cache directory created. Old cache directory deleted.');
+
+            $this->deleteOldCacheDirectory();
+            $io->text('Old directory deleted');
+
+            return 0;
+        } catch (RuntimeException $exception) {
+            $io->error("Error processing {$this->getName()}:\u{a0}" . $exception->getMessage());
+
+            return 1;
+        }
+    }
+
+    private function processChecks(): void
+    {
+        $cache_directory = $this->getCacheDirectoryBasePath();
+        if (!is_writable($cache_directory)) {
+            throw new RuntimeException("Cache directory not writable : [$cache_directory]");
+        }
+    }
+
+    private function renameCurrentCacheDirectory()
+    {
+        $process = new Process(['mv', $this->getCacheDirectoryBasePath(), $this->getCacheDirectoryOldPath()]);
+        $process->run();
+        $this->handleUnsucessfullProcess(__FUNCTION__, $process);
+    }
+
+    private function deleteOldCacheDirectory()
+    {
+        if (file_exists($this->getCacheDirectoryOldPath())) {
+            $process = new Process(['rm', '-rf', $this->getCacheDirectoryOldPath()/*.'/'*/]); // final slash needed
+            $process->run();
+            $this->handleUnsucessfullProcess(__FUNCTION__, $process);
+        }
+    }
+
+    private function createNewCacheDirectory()
+    {
+        $cache_dir = $this->getCacheDirectoryBasePath() . DIRECTORY_SEPARATOR . ((new DebugAdapter())->isDebugModeEnabled() ? 'dev' : 'prod');
+        $process = new Process(['mkdir', $cache_dir, '-p']);
+        $process->run();
+        $this->handleUnsucessfullProcess(__FUNCTION__, $process);
+    }
+
+    /**
+     * @return string Cache directory path without env final directory (eg. dev|prod) without trailing slash
+     *
+     * @throws RuntimeException
+     */
+    private function getCacheDirectoryBasePath(): string
+    {
+        if (!defined('_PS_CACHE_DIR_')) {
+            throw new RuntimeException('Cache directory path not defined in _PS_CACHE_DIR_');
+        }
+
+        return preg_replace('!\\' . DIRECTORY_SEPARATOR . '(prod|dev)\\' . DIRECTORY_SEPARATOR . '$!', '', _PS_CACHE_DIR_);
```

Comment:
I agree that str_replace is simple. But it can't ensure that some part of the path are also replaced.
`_PS_CACHE_DIR_` contains the absolute path to the cache dir, so we have to be careful not to replace part of path not targeted.
So str_replace can have unexpected result on a path like `/home/xavier/dev/boutique-machin/var/cache/dev/`.
So maybe we can use [substr-replace](https://www.php.net/manual/fr/function.substr-replace.php) wich support reverse order replacement.
But will it be more readable ? not sure. A bit faster maybe, but that's probably insignificant.

Reply:
I see but this regex has the same problem no ? since we have the `_PS_CACHE_DIR_` we should use it to find folders inside it instead of guessing things IMO.
Maybe take a look at the sf finder component.

Reply:
No, this problem is not present. Notice the `$` at the end of regexp.
https://3v4l.org/s30X0

Reply:
ye ok seems nice like that then not sure substr will make it more readable in that case